ote_cli/ote_cli/utils/hpo.py

```diff
@@ -0,0 +1,605 @@
+"""
+Utils for HPO with hpopt
+"""
+
+import builtins
+import collections
+import importlib
+import copy
+import multiprocessing
+import os
+import pickle
+import shutil
+import subprocess
+import sys
+import time
+from os import path as osp
+from pathlib import Path
+from typing import Optional
+
+import numpy
+import torch
+import yaml
+from ote_sdk.configuration.helper import create
+from ote_sdk.entities.model import ModelEntity, ModelStatus
+from ote_sdk.entities.model_template import TaskType
+from ote_sdk.entities.subset import Subset
+from ote_sdk.entities.task_environment import TaskEnvironment
+from ote_sdk.entities.train_parameters import TrainParameters, UpdateProgressCallback
+
+from ote_cli.datasets import get_dataset_class
+from ote_cli.utils.importing import get_impl_class
+
+try:
+    import hpopt
+except ImportError as e:
+    print(e)
+    hpopt = None
+
+
+def check_hpopt_available():
+    """Check whether hpopt is avaiable"""
+
+    if hpopt is None:
+        return False
+    return True
+
+
+def get_cuda_device_count():
+    """Count number of avaiable cuda devices"""
+
+    if torch.cuda.is_available():
+        hpo_env = os.environ.copy()
+        cuda_visible_devices = hpo_env.get('CUDA_VISIBLE_DEVICES', None)
+
+        if cuda_visible_devices is None:
+            return torch.cuda.device_count()
+        else:
+            return len(cuda_visible_devices.split(','))
+    return 0
+
+
+def get_cuda_device_list():
+    """Retuns the list of avaiable cuda devices"""
+    if torch.cuda.is_available():
+        hpo_env = os.environ.copy()
+        cuda_visible_devices = hpo_env.get('CUDA_VISIBLE_DEVICES', None)
+
+        if cuda_visible_devices is None:
+            return list(range(0, torch.cuda.device_count()))
+        else:
+            return cuda_visible_devices
```

Comment:
is this list?

Reply:
Thank you for your valuable review. I fixed this bug.